src/Controller/CourseController.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: grantas
+ * Date: 18.3.30
+ * Time: 13.59
+ */
+
+namespace App\Controller;
+use App\Entity\Course;
+use App\Form\CourseType;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+
+class CourseController extends Controller
+{
+
+    /**
+     * @Route("api/course", name="api_course_create", methods="POST")
+     */
+    public function setCourse(Request $request)
+    {
+        $course = new Course();
+        $form = $this->createForm(CourseType::class, $course);
+        $data = json_decode($request->getContent(), true);
+        $form->submit($data);
+
+        if($form->isSubmitted() && $form->isValid()){
+            $course = $form->getData();
+            $course->setCreationDate();
+        }
+        else{
+            $errors = array();
+
+            foreach ($form as $child) {
```

Comment:
Čia įdomus sprendimas... Šiaip turėtų būti tiesiog `$form->getErrors(true)`, tu ten kažką rašei, kad tau tai neveikė, gali gal pabandyt dar ir užmest akį, kodėl taip buvo, nes tas metodas iš esmės panašiai tą ir daro, ką čia darai tu, tik rekursiškai, t.y. tie child'ai irgi gali turėt child'us, jų child'ai irgi gali turėti child'us ir errorai gali slėptis ten, šitaip tu visų neišrinksi. Be to, žiūrint į 41 eilutę, tai paimi tik po pirmą errorą :)